src/Shared/Core/DurableTaskVersionUtil.cs

```diff
@@ -0,0 +1,14 @@
+using System.Reflection;
+
+namespace Microsoft.DurableTask;
+
+public static class DurableTaskUserAgentUtil
+{
+    static string SdkName => "durabletask-dotnet";
+
+    public static string GetUserAgent()
+    {
+        var version = typeof(DurableTaskUserAgentUtil).Assembly.GetName().Version;
```

Comment:
This code gets the assembly version, which is different from the DLL version and isn't really what we want. Use this instead:
```suggestion
        string version = FileVersionInfo.GetVersionInfo(typeof(DurableTaskUserAgentUtil).Assembly.Location).FileVersion;
```

Reply:
Also, make this into a `readonly static` value so that we read it only once and cache it rather than for every gRPC request.

```csharp
static readonly string PackageVersion = FileVersionInfo.GetVersionInfo(typeof(DurableTaskUserAgentUtil).Assembly.Location).FileVersion;
```

Reply:
fixed